nbft.c

```diff
@@ -159,9 +162,36 @@ int discover_from_nbft(nvme_root_t r, char *hostnqn_arg, char *hostid_arg,
 					goto out_free;
 				}
 
+				/* Pause logging for unavailable SSNSs */
+				if ((*ss)->unavailable && verbose < 1) {
+					saved_log_level = nvme_get_logging_level(r,
+										 &saved_log_pid,
+										 &saved_log_tstamp);
+					nvme_init_logging(r, -1, false, false);
+				}
+
 				errno = 0;
 				ret = nvmf_add_ctrl(h, c, cfg);
 
+				/* Resume logging */
+				if ((*ss)->unavailable && verbose < 1)
+					nvme_init_logging(r,
+							  saved_log_level,
+							  saved_log_pid,
+							  saved_log_tstamp);
+
+				/*
+				 * In case this SSNS is marked as 'unavailable',
+				 * make another connection attempt.
```

Comment:
Well this is for discussion, for now all my changes mimic the existing behaviour, i.e. connecting to anything defined. With TP8029 in place we can decide in which circumstances a connection attempt should be or not be made.

Another question outlined in #2219 is about making two-phase connection attempts where unavailable records will only get attempted to be reconnected sometime after switchroot, e.g. via a systemd unit.

Reply:
if the SSNS record is unavailable that's because the primary boot phase set it that way.  I think unavailable records should be retried by nvme-cli.

Yes, we want to make another connection attempt.

Reply:
@tzbatek, I am not suggesting changes to your code.

I was just referring to the comment: "make another connection attempt", but will execute a `continue` and will _not_ make another attempt if the unavailable flag is set, unless I'm misreading the code.

Reply:
Ah, sorry :open_mouth:, right, the comment was slightly confusing, reworded.